app/updatemanager_test.go

```diff
@@ -993,3 +993,50 @@ func TestSaveAndLoadToDB(t *testing.T) {
 		assert.Equalf(t, expiredMap[m.ID], m.expired, "Map with ID %s did not have expected expired status %v", m.ID, expiredMap[m.ID])
 	}
 }
+
+func TestMapUpdates(t *testing.T) {
+	t.Run("TestMapInsert updates", func(t *testing.T) {
+		boot := 100
+		memStore := store.NewMemStore()
+		testMapPool := NewControlMap(memStore, boot)
+		testMapPool.Insert((&UpdateControlMap{
+			ID:       "foo",
+			Priority: 2,
+			States: map[string]UpdateControlMapState{
+				"ArtifactInstall": UpdateControlMapState{
+					Action: "force_continue",
+				},
+			},
+		}).Stamp(1))
+		assert.Eventually(t,
```

Comment:
I think `Eventually` won't work correctly unless you `select` with a `default` case. If the announcement doesn't work, then this will just hang forever.

Reply:
No, it will time out after `2 seconds`, tested it

Reply:
Ok, then there is something I don't understand about `Eventually()`. :-)